src/client/windshaft.js

```diff
@@ -611,16 +614,35 @@ function getAggFN(name) {
     return s.substr(0, s.indexOf('_'));
 }
 
-const endpoint = (conf) => {
-    return `${conf.serverURL}/api/v1/map?api_key=${conf.apiKey}`;
+const endpoint = (conf, path='') => {
+    let url = `${conf.mapsServerURL}/api/v1/map`;
+    if (path) {
+        url += '/' + path;
+    }
+    url = authURL(url, conf);
+    return url;
 };
 
 function getLayerUrl(layergroup, layerIndex, conf) {
     if (layergroup.cdn_url && layergroup.cdn_url.templates) {
         const urlTemplates = layergroup.cdn_url.templates.https;
-        return `${urlTemplates.url}/${conf.username}/api/v1/map/${layergroup.layergroupid}/${layerIndex}/{z}/{x}/{y}.mvt?api_key=${conf.apiKey}`;
+        return authURL(`${urlTemplates.url}/${conf.username}/api/v1/map/${layergroup.layergroupid}/${layerIndex}/{z}/{x}/{y}.mvt`, conf);
+    }
+    return endpoint(conf, `${layergroup.layergroupid}/${layerIndex}/{z}/{x}/{y}.mvt`);
+}
+
+function getSQL(query, conf) {
+    let url = `${conf.sqlServerURL}/api/v2/sql?q=` + encodeURIComponent(query);
+    url = authURL(url, conf);
+    return fetch(url);
```

Comment:
maybe this function could return the `parsed json` so in the places where its used we get rid of the 

```javascript
const response = await getSql(q, c);
const sql = await response.json();
```